snapcraft/storeapi/errors.py

```diff
@@ -170,3 +170,31 @@ class StoreReviewError(StoreError):
     def __init__(self, result):
         self.fmt = self.__messages[result['code']]
         super().__init__()
+
+
+class StoreReleaseError(StoreError):
+
+    __FMT_NOT_REGISTERED = (
+        'Sorry, try `snapcraft register {snap_name}` before trying to '
+        'release or choose an existing revision.')
```

Comment:
Well the error shows up for both and I already requested OLS to provide better errors for when each scenario for a 404 happens